File: Array/Array Nesting/code.js
/** https://leetcode.com/problems/array-nesting/
 * @param {number[]} nums
 * @return {number}
 */
var arrayNesting = function(nums) {
    var max_val=0;
    for(var i=0; i<nums.length; i++){
        var j = i;
        for(var count=0; nums[j]!==-1; count++){
            var temp = j;
            j=nums[j];
            nums[temp]=-1;
        } 
        max_val=Math.max(max_val,count);
    }
    return max_val;
};

File: Array/Sorting Colours/code.py
# https://leetcode.com/problems/sort-colors/
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        def swap(i,j):
            temp=nums[i]
            nums[i]=nums[j]
            nums[j]=temp
            
        lst=[0,0,0]
        for i in range(len(nums)):
            temp=nums[i]
            swap(lst[nums[i]],i)
            if i>0 and nums[i]<nums[i-1]: swap(lst[nums[i]],i)
            for j in range(temp,3):
                lst[j]+=1
            
            
            lst[1]=max(lst[0],lst[1])
            lst[2]=max(lst[0],lst[1],lst[2])
